### PR TITLE
avpops: avp_subst() support pvar as subst parameter

### PR DESCRIPTION
**Pull request by linuxmaniac**
#### Pre-Submission Checklist

- [X] Commit message has the format required by CONTRIBUTING guide
- [X] Commits are split per component (core, individual modules, libs, utils, ...)
- [X] Each component has a single commit (if not, squash them into one commit)
- [X] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [X] Tested changes locally

#### Description
Allow pvar as second parameter of avp_subst() function

### CODE REVIEW

File: src/modules/avpops/avpops.c [hunk 1]
```diff
@@ -1049,9 +1051,36 @@ static int w_op_avps(struct sip_msg* msg, char* param, char *op)
...
+	LM_DBG("%s preparing %s\n", exports.name, subst.s);
+
+	se=subst_parser(&subst);
+	if (se==0)
+	{
+		LM_ERR("%s: bad subst re %s\n", exports.name, subst.s);
+		return E_BAD_RE;
+	}
+
+	return ops_subst(msg, (struct fis_param**)src, se);
```
Comment (miconda) [inline src/modules/avpops/avpops.c]: I think that se needs to be freed, because subst_parser() allocates the result.

Comment (linuxmaniac) [reply src/modules/avpops/avpops.c]: Yes, indeed. Thanks!

**Comment by miconda** [comment]
Looking further at this patch, it requires to have the second parameter as a single variable to be considered with "pvar" support, because I see you search for `/` in the second parameter. It makes it support like:

```
$var(se) = "/abc/xyz/";
avp_subst("$avp(x)", "$var(se)");
```

Right now it support PV in the replacement part of the subst expression, not allowing the match/regexp part.

Maybe is better to just add a new function that will always evaluate the subst parameter with variables, allowing more flexible format for subst parameter. At the end, you just need to have your added code in a dedicated function that evaluates first the parameter and then compiles the subst expression.

So we can have:

```
avp_susbtx("$avp(x)", "/$fU/$rU/g");
```
Allowing also:

```
$var(se) = "/" + $fU + "/" + $rU + "/";
avp_substx("$avp(x)", "$var(se)");
```
The new function can have a different name than avp_substx(), I used to exemplify here.

**Comment by linuxmaniac** [comment]
Added ``avp_subst_pv()`` and now this is supported:
```
    $(avp(src)[*]) = "testME";
    $var(x) = "/e/j/gi";
    avp_subst_pv("$avp(src)", "$var(x)");

    $var(z) = "j";
    $var(y) = "e";
    $var(x) = "/" + $var(y) + "/" + $var(z) + "/gi";
    avp_subst_pv("$avp(src)", "$var(x)");
    avp_subst_pv("$avp(src)", "/$var(y)/$var(z)/gi");
    avp_subst_pv("$avp(src)", "/" + $var(y) + "/" + $var(z) + "/gi");
    avp_subst_pv("$avp(src)", "/e/j/gi");
```

**Comment by henningw** [comment]
Is this is still work in progress or could be merged?

**Comment by linuxmaniac** [comment]
I'm waiting for feedback. No more changes from my side.

**Comment by miconda** [comment]
It is a new function now, it can be merged.

**Pull request merged**